### ISSUE TITLE
Outdated documentation ?

### ISSUE THREAD

**Issue by w0xlt** [rust-bitcoin/rust-secp256k1#412]
Some documentation seems outdated.
For example, the code below, as [described in the documentation](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/key.rs#L196-L198), does not compile:

```rust
use secp256k1::{rand, Secp256k1, SecretKey, KeyPair};

let secp = Secp256k1::new();
let key_pair = KeyPair::new(&secp, &mut rand::thread_rng());
```

The first reason is that `secp256k1::rand` apparently no longer exists.
But even after I changed it to use the `rand`module, an error still persists.

```rust
use secp256k1::{Secp256k1, KeyPair};

fn main() {
    let secp = Secp256k1::new();
    let key_pair = KeyPair::new(&secp, &mut rand::thread_rng());
}
```

```
$ cargo build
Compiling taproot-secp256k1 v0.1.0 (/home/.../test-secp256k1)
error[E0599]: no function or associated item named `new` found for struct `secp256k1::KeyPair` in the current scope
 --> src/main.rs:7:29
  |
7 |     let key_pair = KeyPair::new(&secp, &mut rand::thread_rng());
  |                             ^^^ function or associated item not found in `secp256k1::KeyPair`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `test-secp256k1` due to previous error
```

What is causing the error and how can it be fixed?

**Comment by w0xlt** [comment]
It was solved by adding `features = [ "rand-std" ]` in the `dependencies`:
`secp256k1 = { version = "0.21.3", features = [ "rand-std" ] }`

**Comment by apoelstra** [comment]
Thanks for the report! I think we fixed the documentation in #340 (or one of the other half-dozen doc PRs we merged recently). It should be reflected in the upcoming version, which we hope to release this week.
